src/controllers/universal-router/index.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { Component } from 'react';
+
+import { DEFAULT_HISTORY } from '../../common/constants';
+import { getRouterState, RouterContainer } from '../router-store';
+import { UnlistenHistory } from '../router-store/types';
+import { isNodeEnvironment } from '../../common/utils';
+import { UniversalRouterProps, RequestResourcesParams } from './types';
+import { createMemoryHistory, createLocation } from 'history';
+import { BrowserHistory } from 'src/common/types';
+import { getResourceStore } from '../resource-store';
+import { getRouterStore } from '../router-store';
+
+const getIsStaticRouter = (isStatic?: boolean) =>
+  isStatic === undefined ? isNodeEnvironment() : isStatic;
+
+const getInferredHistory = (history: BrowserHistory, location?: string) =>
+  location ? createMemoryHistory({ initialEntries: [location] }) : history;
+
+/**
+ * Default prop provider for the RouterContainer.
+ *
+ */
+export class UniversalRouter extends Component<UniversalRouterProps> {
+  static defaultProps = {
+    isStatic: false,
```

Comment:
Can you explain again what is `isStatic` for?

Reply:
The prop prevents the router from listening to history. We don’t want or need the router to listen to history in a server context. Perhaps we can fully replace it with the `isNodeEnvironment()` util but let me look into it more and see if there’s any other reason we’d want an explicit prop.